pulp_rpm/app/modulemd.py

```diff
@@ -50,130 +58,124 @@ def modules_packages(modules):
+    new_obsolete[PULP_MODULEOBSOLETES_ATTR.CONTEXT] = obsolete["data"].get("context")
+
+    if obsolete["data"].get("eol_date"):
+        new_obsolete[PULP_MODULEOBSOLETES_ATTR.EOL] = obsolete["data"].get("eol_date")
+    if obsolete["data"].get("obsoleted_by"):
+        new_obsolete[PULP_MODULEOBSOLETES_ATTR.OBSOLETE_BY_MODULE] = obsolete["data"][
+            "obsoleted_by"
+        ].get("module")
+        new_obsolete[PULP_MODULEOBSOLETES_ATTR.OBSOLETE_BY_STREAM] = obsolete["data"][
+            "obsoleted_by"
+        ].get("stream")
+    new_obsolete["snippet"] = snippet
+
+    return new_obsolete
+
+
+def parse_modular(file):
+    """
+    Parse all modular metadata.
     """
-    ret = list()
-    for module in module_names:
-        modulemd = module_index.get_module(module)
-        for obsolete in modulemd.get_obsoletes():
-            new_obsolete = dict()
-            # parsed as in documentation
-            # https://fedora-modularity.github.io/libmodulemd/latest/modulemd-2.0-Modulemd.Obsoletes.html#ModulemdObsoletes--modified
-            new_obsolete["modified"] = datetime.strptime(str(obsolete.props.modified), "%Y%m%d%H%M")
-            new_obsolete["module_name"] = obsolete.props.module_name
-            new_obsolete["module_stream"] = obsolete.props.module_stream
-            new_obsolete["message"] = obsolete.props.message
-            new_obsolete["override_previous"] = obsolete.props.override_previous
-            new_obsolete["module_context"] = obsolete.props.module_context
-            # EOL date is '0' if not specified instead None
-            try:
-                new_obsolete["eol_date"] = datetime.strptime(
-                    str(obsolete.props.eol_date), "%Y%m%d%H%M"
-                )
-            except ValueError:
-                new_obsolete["eol_date"] = None
-            new_obsolete["obsoleted_by_module_name"] = obsolete.props.obsoleted_by_module_name
-            new_obsolete["obsoleted_by_module_stream"] = obsolete.props.obsoleted_by_module_stream
-            # Create a snippet
-            temp_module_index = mmdlib.ModuleIndex.new()
-            temp_module_index.add_obsoletes(obsolete)
-            new_obsolete["snippet"] = temp_module_index.dump_to_string()
-            ret.append(new_obsolete)
-    return ret
+    modulemd_all = []
+    modulemd_defaults_all = []
+    modulemd_obsoletes_all = []
+
+    for module in split_modulmd_file(file):
+        parsed_data = yaml.safe_load(module)
+
+        if parsed_data["document"] == "modulemd":
+            validator = Draft7Validator(MODULEMD_SCHEMA)
+            err = []
+            for error in sorted(validator.iter_errors(parsed_data["data"]), key=str):
+                err.append(error.message)
+            if err:
+                raise ValueError(_("Provided modular data are invalid:'{}'".format(err)))
+            modulemd_all.append(create_modulemd(parsed_data, module))
+        elif parsed_data["document"] == "modulemd-defaults":
```

Comment:
I'm assuming the reason we aren't doing the extra validation for these types is just "level of effort vs. benefit"?  I don't disagree but maybe we should be explicit about it.

Reply:
for modulemd-[defaults|obsoletes] there are just a few fields mandatory (in top level of a tree) so no need big schema validation as in modulemd